src/main/java/net/finmath/fouriermethod/models/BlackScholesModel.java

```diff
@@ -97,4 +126,63 @@ private double getLogDiscountFactorForDiscounting(double time) {
 		return discountCurveForDiscountRate == null ? -discountRate * time : Math.log(discountCurveForDiscountRate.getDiscountFactor(null, time));
 	}
 
+	/**
+	 * @return the referenceDate
+	 */
+	public LocalDate getReferenceDate() {
+		return referenceDate;
```

Comment:
I would suggest the following for "backward compatibility": if referenceDate is null but the model has a discountCurve, then this model should return discountCurve.getReferenceDate().

Reply:
Fine for me, thanks!